keyboards/idobo/keymaps/greenshadowmaker/keymap.c

```diff
@@ -0,0 +1,152 @@
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+#define _ADJUST 16
```

Comment:
Non-consecutive layer numbers will generate extra empty layers in between, unnecessarily increasing flash usage. The better way to define layers is with an enum, eliminating the need to number them manually:

```c
enum layer_names {
    _QWERTY,
    _LOWER,
    _RAISE,
    _ADJUST
};
```

Reply:
added the enum for layer names as suggested and removed the defines...  custom keycode enum had to stay (I don't completely understand this part, I was copying from other keymap)